[Pei Cheng Yi] iP

SeekSaveServe opened the pull request:
### This is a header

- Item 1
- Item 2
- Item 3

1. Number 1
2. Number 2
3. Number 3

> `"Where got time?"` ~ Singaporean Proverb

- [ ] Not done
- [x] Done 

🔥 ✨ 🐍 

#193 is an example of a good repo. Read more [here](https://www.youtube.com/watch?v=dQw4w9WgXcQ) 

`"This is a block quote" Add it with `` `

```python
for i in range(2):
    print("hello world")
```
some text formatting: **bold**, _italic_, ~~stikethrough~~ etc.

SeekSaveServe committed:
Added Level-1

SeekSaveServe committed:
Completed level 2 and refactored duke with switch case

SeekSaveServe committed:
Level 3 completed

SeekSaveServe committed:
Level 4 completed

SeekSaveServe committed:
Completed level 5

SeekSaveServe committed:
:sparkles: Completed level 6

SeekSaveServe committed:
:sparkles: Moved files into packages

SeekSaveServe committed:
:sparkles: Moved files into packages

SeekSaveServe committed:
Merge branch 'master' of https://github.com/SeekSaveServe/CS2103_ip

SeekSaveServe committed:
refactor: Moved retrieveArguments and display into utility classes

SeekSaveServe committed:
:zap: refactor: uses factory methods to handle inputs

The code is verbose and hard to maintain.

Let's refactor each command to its own class.

Let's refactor error handling to each command.

Let's use a factory method to handle inputs instead

SeekSaveServe committed:
:sparkles: Moves all classes into the duke package

SeekSaveServe committed:
:sparkles: Moves all packages into duke package

The code does not move all files in the duke package

Let's move things into the duke package

SeekSaveServe committed:
:sparkles: feature: adds deadline formatting and reading

Currently deadlines are read as String with no verification.

We should parse the deadline as a string.

We should check that the string adheres to the proper format

SeekSaveServe committed:
Merge pull request #1 from SeekSaveServe/branch-level-8

Branch level 8

SeekSaveServe committed:
:sparkles: adds save feature to duke

Current data is lost upon exiting duke.

We should save the tasks in a local file so that they can be retrieved.

SeekSaveServe committed:
Merge pull request #2 from SeekSaveServe/branch-leve-7

:sparkles: adds save feature to duke

SeekSaveServe committed:
:sparkles: refactors code into Parser, Ui, Storage

Currently the code lacks oop

Therefore lets add more oop :fire:

SeekSaveServe committed:
adds gradle

Currently gradle was not available for us to autoate things

We should use gradle

SeekSaveServe committed:
adds checkstyle :fire:

Currently there is no checkstyle configured

Let's fix that

SeekSaveServe committed:
adds test files

SeekSaveServe committed:
adds test cases for gradle

There was no unit testing done

Let's add unit tests in

SeekSaveServe committed:
adds test cases for gradle

There was no unit testing done

Let's add unit tests in

SeekSaveServe committed:
adds Javadoc to documents without javadoc

We do not have enough javadoc

Let's add more

SeekSaveServe committed:
Merge pull request #3 from SeekSaveServe/A-Javadoc

adds Javadoc to documents without javadoc

SeekSaveServe committed:
:sparkle: aligns code with coding convention

Currently, the code violates coding  convention

Let's fix that

SeekSaveServe committed:
Merge pull request #4 from SeekSaveServe/A-CodingStandard

:sparkle: aligns code with coding convention

SeekSaveServe committed:
:sparkles: feature: adds find feature

Currently, no find feature

Let's fix that

SeekSaveServe committed:
Merge pull request #5 from SeekSaveServe/Level-9

:sparkles: feature: adds find feature

SeekSaveServe committed:
:sparkles: feature: adds GUI

Currently, no GUI

Let's fix that

SeekSaveServe committed:
:sparkles: feature: adds GUI

Currently, no GUI

Let's fix that

SeekSaveServe committed:
:sparkles: feature: adds GUI

Currently, no GUI

Let's fix that

SeekSaveServe committed:
Merge pull request #6 from SeekSaveServe/Level-10

Level 10

SeekSaveServe committed:
:fire: adds Checkstyle

Currently, have checkstyle

But no branch

In short need to make new branch

SeekSaveServe committed:
Merge pull request #7 from SeekSaveServe/A-CheckStyle

:fire: adds Checkstyle

SeekSaveServe committed:
:sparkles: uses varargs

Currently, have varargs

But no branch

In short need to make new branch

SeekSaveServe committed:
Merge pull request #8 from SeekSaveServe/A-Varargs

:sparkles: uses varargs

SeekSaveServe committed:
removes duplicate files

SeekSaveServe committed:
Merge pull request #9 from SeekSaveServe/A-Varargs

removes duplicate files

SeekSaveServe committed:
:fire: Creates jar files

Currently we don't have jar file

Let's fix that

SeekSaveServe committed:
:fire: Creates jar files

Currently we don't have jar file

Let's fix that

SeekSaveServe committed:
Merge pull request #10 from SeekSaveServe/A-Varargs

A varargs

SeekSaveServe committed:
:sparkles: Adds comments explaining why Jar is the entrypoint

SeekSaveServe committed:
:sparkles: Adds comments explaining why Jar is the entrypoint

SeekSaveServe committed:
Update README.md

adds picture of DUKE with GUI

SeekSaveServe committed:
Update README.md

mqchua commented on src/main/java/duke/Gui/DialogBox.java:
```diff
@@ -0,0 +1,49 @@
+    private ImageView displayPicture;
+
+    public DialogBox(Label l, ImageView iv) {
+        text = l;
```
I think variable "L" for text can be more explicit

mqchua commented on src/main/java/duke/commands/AddTodoCommand.java:
```diff
@@ -0,0 +1,39 @@
+ * Adds Todo to the tasklist
+ */
+public class AddTodoCommand extends ShowList {
+    protected String descrition;
```
Mini typo over here

mqchua commented on src/main/java/duke/lists/TaskList.java:
```diff
@@ -0,0 +1,110 @@
+            output += (i + 1);
+            output += " ";
+            output += tasks.get(i);
+            output += "\n";
```
Is it clearer to condense it into one line? 🤔

mqchua commented on src/main/java/duke/utils/Storage.java:
```diff
@@ -0,0 +1,203 @@
+            put("December", "12");
+            put("November", "11");
+            put("October", "10");
+            put("September", "09");
```
I think can be clearer if named in order of the month, with December at the end 😄

mqchua reviewed (commented):
Overall great PR! Just some minor nit picks 😄

SeekSaveServe committed:
:speech_balloon: Add assertions to document assumptions

Important assumptions in the code are not documented

Assertions help to document assumptions

Let's,
* Add assertions
* Incorporate this into our future code

SeekSaveServe committed:
:writing_hand: Fix code quality

There are minor code quality issues

Code quality is important for a healthy and easy to read code base

Let's,
* Fix code quality
* Incorporate this into our future code

SeekSaveServe committed:
Merge pull request #11 from SeekSaveServe/A-Assertions

A-assertions, good job!

SeekSaveServe committed:
Merge pull request #12 from SeekSaveServe/A-CodeQuality

:writing_hand: Fix code quality

SeekSaveServe committed:
:fire: Feature: Add reminder feature

There is no feature to remind the user of the next upcoming task

It is important for Duke to inform users of upcoming tasks

Let's,
* Add in this feature
* Don't forget to change the read me

SeekSaveServe committed:
:fire: Feature: Add reminder feature

There is no feature to remind the user of the next upcoming task

It is important for Duke to inform users of upcoming tasks

Let's,
* Add in this feature
* Don't forget to change the read me

SeekSaveServe committed:
Merge pull request #13 from SeekSaveServe/BCD-Extension_B-Reminder_C-BetterSearch

Bcd extension b reminder c better search

doimoiboi commented on src/main/java/duke/commands/ShowList.java:
```diff
@@ -0,0 +1,29 @@
+/**
+ * Displays the entire list of existing tasks
+ */
+public class ShowList extends DisplayCommand {
```
Any reason this class is named as a verb instead of a noun?

doimoiboi commented on src/main/java/duke/entities/Deadline.java:
```diff
@@ -0,0 +1,20 @@
+package duke.entities;
+
+import java.time.LocalDateTime;
+
+import duke.exceptions.DukeException;
```
I like how you got the standard for these statements right! Easy to mess it up!

doimoiboi reviewed (commented):
Follows the coding standard very well! LGTM!

doimoiboi commented on plugin/src/main/java/duke/commands/ShowList.java:
```diff
@@ -0,0 +1,29 @@
+/**
+ * Displays the entire list of existing tasks
+ */
+public class ShowList extends DisplayCommand {
```
Is there any reason to name this class as a verb instead of a noun?

doimoiboi commented on plugin/src/main/java/duke/entities/Deadline.java:
```diff
@@ -0,0 +1,20 @@
+
+    @Override
+    public String toString() {
+        String marker = isDone() ? "[X] " : "[ ] ";
```
I do think using the ternary operator here makes the code much more succinct. However perhaps using if-else loops would be more consistent with the overall design? Furthermore not everyone might agree the former is more readable. I confess this is mostly personal preference though.

doimoiboi commented on plugin/src/main/java/duke/enums/Commands.java:
```diff
@@ -0,0 +1,32 @@
+ * @author Pei Cheng Yi A029823Y
+ */
+
+public enum Commands {
+    EXIT("bye"),
+    LIST("list"),
+    GREET("greet"),
+    TASK("Add Task"),
+    TODO("todo"),
+    EVENT("event"),
+    DEADLINE("deadline"),
+    DELETE("delete"),
+    MARK("mark"),
+    FIND("find"),
+    REMINDER("remind"),
+    ERROR("");
```
I like the use of enums especially for the duke commands. Great work!

doimoiboi reviewed (commented):
LGTM! Code is really organised and clean.

SeekSaveServe committed:
:sparkles: Add CI with gradle.yml

Currently there's no CI

Let's add CI

SeekSaveServe committed:
Update README.md

SeekSaveServe committed:
Update README.md

SeekSaveServe committed:
Update README.md

SeekSaveServe committed:
:sparkles: Add Ui.png to showcase product and updated README

Currently, there is no Ui.png to showcase

Let's add Ui.png

SeekSaveServe committed:
:sparkles: Add Ui.png to showcase product and updated README

Currently, there is no Ui.png to showcase

Let's add Ui.png

SeekSaveServe committed:
:bug: Fix Jar bug

Currently Jar file does not run properly in a new directory as the folder 'data' is not created

Let's fix that by making storage create the 'data' folder before creating the duke.txt storage file
